azkaban-common/src/main/java/azkaban/project/DirectoryYamlFlowLoader.java

```diff
@@ -217,4 +222,22 @@ private void addEdges(final AzkabanNode node, final AzkabanFlow azkabanFlow,
     }
   }
 
+  private void setConditionOnJobStatus(final Node node) {
+    String condition = node.getCondition();
+    if (condition != null) {
+      final String patternString =
```

Comment:
Maybe add a comment to show what pattern should be correct, and what are wrong.

Reply:
Sure, will add the comment.